### PR TITLE
ui v2: select

### PR DESCRIPTION
**Pull request by danielhochman**
### Description
Implement the v2 Figma designs.

### Testing Performed
Storybook.

### TODO
- [ ] style popover menu in followup

### CODE REVIEW

File: frontend/packages/core/src/Input/select.tsx [hunk 1]
```diff
@@ -1,60 +1,163 @@
...
+  "&.Mui-focused": {
+    borderColor: "#3548d4",
+  },
+
+  "&.Mui-error": {
+    borderColor: "#db3615"
+  },
+
+  ".MuiSelect-root": {
+    padding: "14px 16px",
```
Comment (dschaller) [inline frontend/packages/core/src/Input/select.tsx:94]:
```suggestion
    padding: "14px 16px",
    borderRadius: "3px",
```

File: frontend/packages/core/src/Input/select.tsx [hunk 2]
```diff
@@ -1,60 +1,163 @@
...
+  ".MuiSelect-icon.Mui-disabled > svg": {
+    color: "rgba(13, 16, 48, 0.6)"
+  },
+
+  ".MuiSelect-icon > svg": {
+    color: "rgba(13, 16, 48, 0.6)",
+    position: "absolute",
+    left: "50%",
+    top: "50%",
+    transform: "translate(-50%, -50%)"
```
Comment (dschaller) [inline frontend/packages/core/src/Input/select.tsx:120]:
```suggestion
```

File: frontend/packages/core/src/Input/select.tsx [hunk 3]
```diff
@@ -1,60 +1,163 @@
...
+
+  ".MuiSelect-icon": {
+    borderLeftColor: "inherit",
+    borderLeftWidth: "1px",
+    borderLeftStyle: "solid",
+    height: "100%",
+    width: "48px",
+    top: "unset",
+    transform: "unset",
+  },
```
Comment (dschaller) [inline frontend/packages/core/src/Input/select.tsx:109]:
```suggestion
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
  },
```

File: frontend/packages/core/src/Input/select.tsx [hunk 4]
```diff
@@ -1,60 +1,163 @@
...
+    top: "50%",
+    transform: "translate(-50%, -50%)"
+  },
+
+  "&.Mui-focused > .MuiSelect-icon > svg": {
+    color: "#0d1030",
+  },
+
+  ".MuiSelect-icon.MuiSelect-iconOpen > svg": {
+    transform: "translate(-50%, -50%) rotate(180deg)",
```
Comment (dschaller) [inline frontend/packages/core/src/Input/select.tsx:128]:
```suggestion
    transform: "rotate(180deg)",
```

File: frontend/packages/core/src/Input/stories/select.stories.tsx [hunk 5]
```diff
@@ -13,46 +13,51 @@ export default {
   },
 } as Meta;
 
-const Template = (props: SelectProps) => <Select name="demo" {...props} />;
+const Template = (props: SelectProps) => <Select name="storybookDemo" {...props} />;
 
-export const Primary = Template.bind({});
-Primary.args = {
+export const Basic = Template.bind({});
+Basic.args = {
```
Comment (dschaller) [inline frontend/packages/core/src/Input/stories/select.stories.tsx:19]: Can we call this Primary?

File: frontend/packages/core/src/Input/select.tsx [hunk 6]
```diff
@@ -74,22 +177,20 @@ const Select: React.FC<SelectProps> = ({
...
-        fullWidth
       >
         {options.map(option => (
           <MenuItem key={option.label} value={option?.value || option.label}>
             {option.label}
           </MenuItem>
         ))}
       </StyledSelect>
-    </FormControl>
+      {helperText && <StyledFormHelperText>{error && <ErrorIcon />}{helperText}</StyledFormHelperText>}
```
Comment (dschaller) [inline frontend/packages/core/src/Input/select.tsx:192]: I think you'll probably carry over the fix from the text field but just so I don't forget the error icon should be centered here.

File: frontend/packages/core/src/Input/select.tsx [hunk 7]
```diff
@@ -1,60 +1,164 @@
...
+    alignItems: "center",
+    boxSizing: "border-box",
+  },
+
+  ".MuiSelect-icon.Mui-disabled > svg": {
+    color: "rgba(13, 16, 48, 0.6)",
+  },
+
+  ".MuiSelect-icon > svg": {
+    color: "rgba(13, 16, 48, 0.6)",
```
Comment (dschaller) [inline frontend/packages/core/src/Input/select.tsx:121]: It looks like this was updated to remove the transparency

File: frontend/packages/core/src/Input/select.tsx [hunk 8]
```diff
@@ -1,60 +1,163 @@
...
+};
+
+
+const BaseSelect = ({ ...props }: MuiSelectProps) => (
+  <MuiSelect
+    disableUnderline
+    fullWidth
+    IconComponent={SelectIcon}
+    MenuProps={{
+      style: { marginTop: "2px" },
```
[the remark this answers is not in the document]
Comment (dschaller) [reply frontend/packages/core/src/Input/select.tsx:67]: Did you want to fix this as part of a subsequent PR when we tackle the menu?

File: frontend/packages/core/src/Input/stories/select.stories.tsx (hunk 5 above)
Comment (dschaller) [inline frontend/packages/core/src/Input/stories/select.stories.tsx:19]:
```suggestion
export const Primary = Template.bind({});
Primary.args = {
```

File: frontend/packages/core/src/Input/stories/select.stories.tsx [hunk 9]
```diff
@@ -13,46 +13,51 @@ export default {
...
     {
-      label: "option 2",
+      label: "Option 2",
     },
   ],
 };
 
+export const Disabled = Template.bind({});
+Disabled.args = {
+  ...Basic.args,
```
Comment (dschaller) [inline frontend/packages/core/src/Input/stories/select.stories.tsx:33]:
```suggestion
  ...Primary.args,
```

File: frontend/packages/core/src/Input/stories/select.stories.tsx [hunk 10]
```diff
@@ -13,46 +13,51 @@ export default {
...
+  disabled: true,
+};
+
+export const Error = Template.bind({});
+Error.args = {
+  ...Basic.args,
```
Comment (dschaller) [inline frontend/packages/core/src/Input/stories/select.stories.tsx:39]:
```suggestion
  ...Primary.args,
```

File: frontend/packages/core/src/Input/stories/select.stories.tsx [hunk 11]
```diff
@@ -13,46 +13,51 @@ export default {
...
+  error: true,
+  helperText: "There was a problem!",
+};
+
 export const CustomValues = Template.bind({});
 CustomValues.args = {
+  ...Basic.args,
```
Comment (dschaller) [inline frontend/packages/core/src/Input/stories/select.stories.tsx:46]:
```suggestion
  ...Primary.args,
```

File: frontend/packages/core/src/Input/stories/select.stories.tsx [hunk 12]
```diff
@@ -13,46 +13,51 @@ export default {
...
-  },
-};
-WithLabel.args = {
-  ...Primary.args,
-  defaultOption: 0,
-  label: "Please select one",
-  maxWidth: "100px",
+export const WithoutLabel = Template.bind({});
+WithoutLabel.args = {
+  ...Basic.args,
```
Comment (dschaller) [inline frontend/packages/core/src/Input/stories/select.stories.tsx:61]:
```suggestion
  ...Primary.args,
```

File: frontend/packages/core/src/Input/select.tsx (hunk 8 above)
[the remark this answers is not in the document]
Comment (danielhochman) [reply frontend/packages/core/src/Input/select.tsx:67]: yeah

**Review by dschaller — APPROVED** [review]

**Pull request merged**